operations/helm/charts/mimir-distributed/values.yaml

```diff
@@ -1097,7 +1097,8 @@ ruler:
       memory: 128Mi
 
   # Additional ruler container arguments, e.g. log level (debug, info, warn, error)
-  extraArgs: {}
+  extraArgs:
+    distributor.remote-timeout: 10s
```

Comment:
I've been told by @krajorama in other PR to not keep `extraArgs` empty and just set it right after `- "-config.file=/etc/mimir/mimir.yaml"` in `operations/helm/charts/mimir-distributed/templates/ruler/ruler-dep.yaml`. If an user wants to override it, then setting it in `extraArgs` will take precedence.

Reply:
i agree. it also means this change will take effect if the user already has some other parameters configured in `extraArgs`

Reply:
Good to know, thanks for the tip!

Just for my own edification - I'm a little unclear on how how the override would work. If the user adds the `-distributor.remote-timeout` flag to `extraArgs`, wouldn't that flag then appear in the ruler's args list twice?

Reply:
> wouldn't that flag then appear in the ruler's args list twice?

that's right. Go's default behaviour is to take the value from the last usage of the flag. I couldn't find docs, but you can use this go program to prove verify it experimentally (by running all possible values of the flag in all possible orders? idk) https://go.dev/play/p/Qq2uqAHiFuQ


```go
package main

import (
	"flag"
	"fmt"
)

func main() {
	var f string
	flag.StringVar(&f, "flag", "", "help message for flagname")
	flag.Parse()
	fmt.Println("f:", f)
}
```

```
go run main.go -flag=123 -flag=qwerty
f: qwerty
```

Reply:
Gotcha - thanks Dimitar!